src/Entities/Entity.cpp

```diff
@@ -238,6 +238,40 @@ void cEntity::TakeDamage(eDamageType a_DamageType, cEntity * a_Attacker, int a_R
 
 
 
+void cEntity::TakeDamage(eDamageType a_DamageType, UInt32 a_AttackerID, int a_RawDamage, double a_KnockbackAmount)
+{
+	cPawn * Attacker = nullptr;
+	class cNotifyWolves : public cEntityCallback
+	{
+	public:
+		cPawn * m_Attacker;
+
+		virtual bool Item(cEntity * a_Attacker) override
+		{
+			if (a_Attacker->IsPawn())
+			{
+				m_Attacker = static_cast<cPawn*>(a_Attacker);
+			}
+			else
+			{
+				m_Attacker = nullptr;
+			}
+			return true;
+		}
+	} Callback;
+
+
+	m_World->DoWithEntityByID(a_AttackerID, Callback);
+	Attacker = Callback.m_Attacker;
```

Comment:
Maybe. It depends on the thread and lock guarantees that DoTakeDamage gets from its caller.
But it would be safer to do the cEntity::TakeDamage call from the callback. It might be easier to add a std::function overload to cWorld::DoWithEntityByID, and use lambdas than do manual captures in cNotifyWolves.